MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java

```diff
@@ -0,0 +1,504 @@
+     */
+    public void removeFormerSpouse(FormerSpouse formerSpouse) {
+        getFormerSpouses().remove(formerSpouse);
+    }
+
+    /**
+     * @return the family map for this person
+     */
+    public Map<FamilialRelationshipType, List<UUID>> getFamily() {
+        return family;
+    }
+
+    /**
+     * This is used to add a new family member
+     * @param relationshipType the relationship type between the two people
+     * @param id the id of the person to add
+     */
+    public void addFamilyMember(FamilialRelationshipType relationshipType, UUID id) {
+        if (id != null) {
+            getFamily().putIfAbsent(relationshipType, new ArrayList<>());
+            getFamily().get(relationshipType).add(id);
+        }
+    }
+
+    /**
+     * This is implemented for future use, as it will be required for the family tree implementation
+     * @param relationshipType the FamilialRelationshipType of the person to remove
+     * @param id the id of the person to remove
+     */
+    public void removeFamilyMember(FamilialRelationshipType relationshipType, UUID id) {
+        if (getFamily().get(relationshipType) == null) {
+            MekHQ.getLogger().error(getClass(), "removeFamilyMember",
+                    "Could not remove unknown family member of relationship "
+                            + relationshipType.name() + " and UUID " + id.toString() + ".");
+            return;
+        }
+
+        List<UUID> familyTypeMembers = getFamily().get(relationshipType);
+        familyTypeMembers.remove(id);
+        if (familyTypeMembers.isEmpty()) {
+            getFamily().remove(relationshipType);
+        }
+    }
+    //endregion Getters/Setters
+
+    //region Boolean Checks
+    /**
+     * @return true if the person has either a spouse, any children, or specified parents.
+     *          These are required for any extended family to exist.
+     */
+    public boolean hasAnyFamily() {
+        return hasChildren() || hasSpouse() || hasParents();
+    }
+
+    /**
+     * @return true if the person has a spouse, false otherwise
+     */
+    public boolean hasSpouse() {
+        return (getSpouseId() != null);
+    }
+
+    /**
+     * @return true if the person has a former spouse, false otherwise
+     */
+    public boolean hasFormerSpouse() {
+        return !getFormerSpouses().isEmpty();
+    }
+
+    /**
+     * @return true if the person has at least one kid, false otherwise
+     */
+    public boolean hasChildren() {
+        return (getFamily().get(FamilialRelationshipType.CHILD) != null);
+    }
+
+    /**
+     * @return true if the Person has any parents, otherwise false
+     */
+    public boolean hasParents() {
+        return (getFamily().get(FamilialRelationshipType.PARENT) != null);
+    }
+
+    // Recursive Search Format to determine mutual ancestry
+    public boolean checkMutualAncestors(Person person, Campaign campaign) {
```

Comment:
I think this is ripe for unit tests, along with removing relations to make sure the tree is pruned.

Reply:
And while you're at it, let's put some comments on these public methods, as I don't really have any idea what they do just looking at the name.